.github/workflows/test.yml

```diff
@@ -4,7 +4,7 @@ jobs:
   test:
     strategy:
       matrix:
-        go-version: [1.19.x]
+        go-version: [1.21.x]
```

Comment:
Why is the version in square brackets in `test.yml`?

Reply:
As far as I understand, these repositories are libraries only and are never deployed so that the testing workflow is not done via Jenkins but via github workflows directly, so that a `test.yml` file is required and a special format seems to be required for this (I suppose that there are specific cases where one would need multiple go versions to run tests for some specific repositories ? :thinking:) .

Reply:
Oh I get it a bit better now. We are using a [matrix strategy](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) (see [this line](https://github.com/edgelaboratories/interpolator/blob/08e6fbd47465a4703ec779ad23a70ca574c0b61b/.github/workflows/test.yml#L6)). We can (although we don't) enter multiple values for the go version and for the platform. The bracket designates an array.

Reply:
I don't understand why the fact that this repo is a library and is not deployed means that we cannot (or don't want to) use jenkins and prefer github actions. Do you know why? Overall, I don't know what drives the choice between github actions and jenkins (note that one can do the entire CI/CD with github actions [incl. deployment](https://docs.github.com/en/github-ae@latest/actions/deployment/about-deployments/deploying-with-github-actions)).

Reply:
No idea, this might have been a legacy choice